exercises/edward_brown/ex01_basics/src/WordCharPredicates.cpp

```diff
@@ -0,0 +1,5 @@
+#include <cctype>
+
+bool isWordChar(char Candidate) { return std::isspace(Candidate) == 0; }
+
+bool isWhitespaceChar(char Candidate) { return std::isspace(Candidate) != 0; }
```

Comment:
I was a little confused by the wording of the sentence regarding hyphenation, specifically:
> treat hypenated words as a different word.

Taking the following as an example input `hello-world hello world`, which of the two (or neither) is/was the desired behaviour.

Hyphenated words are treated as different word**s**  producing
```
hello   2
world   2
```

or

Hyphenated words are treated as a different word (which is different from those which form it).
```
hello-word   1
hello        1
world        1
```
At the moment I believe the program exhibits the second of these behaviours since `-` is not a white space character and neither is it a 'punctuation mark' which means it won't be stripped out when inserting into the map. 

I would guess switching to the second behaviour could be achieved by adding a new function 
```
bool isWhitespaceOrHyphenChar(char Candidate) {
  return isWhitespaceChar(candidate) || candidate == '-'; 
}
``` 
and is used where `isWhitespaceChar` is currently used.